lua/plenary/luascan.lua

```diff
@@ -0,0 +1,128 @@
+local m = {}
+
+local read_file = function(filepath)
+  local fd = vim.loop.fs_open(filepath, "r", 438)
```

Comment:
It may be better for performance to first check with fs_stat then what type it is then open.

Reply:
Could this be a better version of reading file content?

```lua
function read_file(path)
  local stat,content
  stat = vim.loop.fs_stat(path)
  if stat and stat.type == "directory" then return "" end
  path = io.open(path, "rb")
  content = path:read("*all")
  path:close()
  return vim.split(content, "\n")
end
```

Reply:
If skipping the call to read_file on some entries may improve performance, then using https://keplerproject.github.io/luafilesystem/examples.html or something similar with vim.loop.fs_stat, to create a list of a pair of paths and their type in scan_dir might be a good idea. 

FYI: it turn out that any neovim lua package can be used by the user. (not sure if this worked for me becuase of planery installing luefilesystem though.

Reply:
Right now this only is a `find` replacement not a grep. `read_file` currently only exists because i was too lazy to include `Path` module.
Also i don't see how `io.open` could be a better read then `vim.loop.fs_`